OpenMage/magento-lts
# Fix integers being cast to decimal in some particular cases

### Description (*)

This provides a fix for the issue #1192 that I have created.
Details about the fix can be found here: https://magento.stackexchange.com/questions/105244/magento-1-admin-product-grid-column-value-disappear-on-name-search/105421

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#1192

### Manual testing scenarios (*)

1. Run the following script with catalog product flat disabled
```<?php

require_once 'app/Mage.php';

Mage::app();

$collection = Mage::getModel('catalog/product')->getCollection()
    ->addAttributeToSelect('tax_class_id')
    ->addAttributeToSort('name');

// toggle this to see the weird result
$collection->addAttributeToSelect('price');

// to make the script faster and lighter
$collection->setPageSize(1);

if ($collection->getSize() > 0) {
    var_dump($collection->getFirstItem()->getData('tax_class_id'));
}
```
2. Notice the bug.
3. Apply the fix.
4. Run the script again.

### Questions or comments


### Contribution checklist (*)
